Bug Report - TypeScript error for renderRow, renderLeft/RightButton

# Describe the bug

```
  node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.d.ts:409:3
    409   renderRow?: (data: GooglePlaceData) => React.ComponentType<{}>;
          ~~~~~~~~~
    The expected type comes from property 'renderRow' which is declared here on type 'IntrinsicAttributes & GooglePlacesAutocompleteProps & RefAttributes<GooglePlacesAutocompleteRef>'
```

a similar error manifests for all three renderRow renderLeftButton renderRightButton.

```
src/routes/LiveUserProfile/LiveUserLocationModal/index.tsx:200:21 - error TS2322: Type '() => JSX.Element' is not assignable to type '() => ComponentType<{}>'.
  Type 'Element' is not assignable to type 'ComponentType<{}>'.
    Type 'Element' is not assignable to type 'FunctionComponent<{}>'.
      Type 'Element' provides no match for the signature '(props: { children?: ReactNode; }, context?: any): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | null'.

200                     renderLeftButton={() => {
                        ~~~~~~~~~~~~~~~~

  node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.d.ts:407:3
    407   renderLeftButton?: () => React.ComponentType<{}>;
          ~~~~~~~~~~~~~~~~
    The expected type comes from property 'renderLeftButton' which is declared here on type 'IntrinsicAttributes & GooglePlacesAutocompleteProps & RefAttributes<GooglePlacesAutocompleteRef>'

src/routes/LiveUserProfile/LiveUserLocationModal/index.tsx:208:21 - error TS2322: Type '() => JSX.Element' is not assignable to type '() => ComponentType<{}>'.
  Type 'Element' is not assignable to type 'ComponentType<{}>'.
    Type 'Element' is not assignable to type 'FunctionComponent<{}>'.

208                     renderRightButton={() => {
                        ~~~~~~~~~~~~~~~~~

  node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.d.ts:408:3
    408   renderRightButton?: () => React.ComponentType<{}>;
          ~~~~~~~~~~~~~~~~~
    The expected type comes from property 'renderRightButton' which is declared here on type 'IntrinsicAttributes & GooglePlacesAutocompleteProps & RefAttributes<GooglePlacesAutocompleteRef>'

src/routes/LiveUserProfile/LiveUserLocationModal/index.tsx:224:21 - error TS2322: Type '(results: GooglePlaceData) => Element' is not assignable to type '(data: GooglePlaceData) => ComponentType<{}>'.
  Type 'Element' is not assignable to type 'ComponentType<{}>'.
    Type 'Element' is not assignable to type 'FunctionComponent<{}>'.

224                     renderRow={(results): JSX.Element => {
                        ~~~~~~~~~
```
# Reproduction - (required - issue will be closed without this)

Steps to reproduce the behavior - a minimal reproducible code example, link to a [snack](https://snack.expo.io) or a repository.

**Please provide a FULLY REPRODUCIBLE example.**

<details>
  <summary>Click to expand!</summary>
  
  ```javascript
     <GooglePlacesAutocomplete
                    renderLeftButton={() => {
                      return (
                        <SVGIcon
                          icon={icons.search.searchMagnifyingGlass}
                          style={styles.searchIcon}
                        />
                      );
                    }}
                    renderRightButton={() => {
                      return (
                        <>
                          <TouchableOpacity
                            onPress={(): void => {
                              onDismissModal(false);
                            }}
                          >
                            <Text style={styles.cancelText}>
                              {locales.cancel}
                            </Text>
                          </TouchableOpacity>
                        </>
                      );
                    }}
                    renderRow={(results): JSX.Element => {
                      return (
                        <View style={styles.resultsContainer}>
                          <TouchableOpacity
                            onPress={(): void => {
                              setLocation(results.description);
                            }}
                          >
                            <View style={styles.resultsRow}>
                              <Text style={styles.resultsText}>
                                {results.description}
                              </Text>
                              <SVGIcon
                                icon={icons.search.caretOut}
                              />
                            </View>
                          </TouchableOpacity>
                        </View>
                      );
                    }}
                    query={{
                      components: 'country:us',
                      key: 'myApiKey',
                      language: 'en',
                      types: '(cities)',
                    }}
                  />
    
  ```
</details>

_Please remember to remove you google API key from the code you provide here_

# Additional context

Obviously this appears to be a TS issue. 

Looking here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts

I see this:

```
    type ElementType<P = any> =
        {
            [K in keyof JSX.IntrinsicElements]: P extends JSX.IntrinsicElements[K] ? K : never
        }[keyof JSX.IntrinsicElements] |
        ComponentType<P>;
    /**
     * @deprecated Please use `ElementType`
     */
    type ReactType<P = any> = ElementType<P>;
    type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;

```



- Library Version: 2.0.3
- React Native Version: 0.61.5
- TypeScript Version: 3.8.2

- [ ] iOS
- [ ] Android
- [ ] Web

If you are using expo please indicate here:

- [ ] I am using expo

Add any other context about the problem here, screenshots etc

--- a/GooglePlacesAutocomplete.d.ts
+++ b/GooglePlacesAutocomplete.d.ts
@@ -401,7 +401,7 @@ interface GooglePlacesAutocompleteProps {
   predefinedPlaces?: Place[];
   predefinedPlacesAlwaysVisible?: boolean;
   preProcess?: (text: string) => string;
-  query: Query;
+  query: Query | Object;
   renderDescription?: (description: DescriptionRow) => string;
   renderHeaderComponent?: () => JSX.Element | React.ComponentType<{}>;
   renderLeftButton?: () => JSX.Element | React.ComponentType<{}>;
@@ -410,11 +410,11 @@ interface GooglePlacesAutocompleteProps {
   
   // sets the request URL to something other than the google api.  Helpful if you want web support or to use your own api.
   requestUrl?: RequestUrl;
-  styles?: Partial<Styles>;
+  styles?: Partial<Styles> | Object;
   suppressDefaultStyles?: boolean;
   textInputHide?: boolean;
   // text input props
-  textInputProps?: TextInputProps;
+  textInputProps?: TextInputProps | Object;
   timeout?: number;
 }
 

--- a/GooglePlacesAutocomplete.d.ts
+++ b/GooglePlacesAutocomplete.d.ts
@@ -386,7 +386,7 @@ interface GooglePlacesAutocompleteProps {
   isRowScrollable?: boolean;
   keyboardShouldPersistTaps?: 'never' | 'always' | 'handled';
   // use the ListEmptyComponent prop when no autocomplete results are found.
-  listEmptyComponent?: React.ComponentType<{}>;
+  listEmptyComponent?: JSX.Element | React.ComponentType<{}>;
   listUnderlayColor?: string;
   listViewDisplayed?: 'auto' | boolean;
   minLength?: number; // minimum length of text to search
@@ -403,10 +403,11 @@ interface GooglePlacesAutocompleteProps {
   preProcess?: (text: string) => string;
   query: Query;
   renderDescription?: (description: DescriptionRow) => string;
-  renderHeaderComponent?: () => React.ComponentType<{}>;
-  renderLeftButton?: () => React.ComponentType<{}>;
-  renderRightButton?: () => React.ComponentType<{}>;
-  renderRow?: (data: GooglePlaceData) => React.ComponentType<{}>;
+  renderHeaderComponent?: () => JSX.Element | React.ComponentType<{}>;
+  renderLeftButton?: () => JSX.Element | React.ComponentType<{}>;
+  renderRightButton?: () => JSX.Element | React.ComponentType<{}>;
+  renderRow?: (data: GooglePlaceData) => JSX.Element | React.ComponentType<{}>;
+  
   // sets the request URL to something other than the google api.  Helpful if you want web support or to use your own api.
   requestUrl?: RequestUrl;
   styles?: Partial<Styles>;

